src/MarDyn.cpp

```diff
@@ -132,8 +132,18 @@ int run_unit_tests(const Values &options, const std::vector<std::string> &args)
  */
 int main(int argc, char** argv) {
 #ifdef ENABLE_MPI
+#if defined(_OPENMP)
+	int thread_support_level_requested = MPI_THREAD_FUNNELED; /* So far only the main thread makes MPI calls */
+	int thread_support_level_provided = MPI_THREAD_SINGLE;
+	MPI_Init_thread(&argc, &argv, thread_support_level_requested, &thread_support_level_provided);
+	if (thread_support_level_provided < thread_support_level_requested) {
+		std::cerr << "Error: MPI implementation does not support the requested thread support level." << std::endl;
+		std::exit(EXIT_FAILURE);
```

Comment:
Shouldn't we still call some `MPI` cleanup function? How about:
```suggestion
		mardyn_exit(EXIT_FAILURE);
```